### PR TITLE
MPRIS support

### PR DESCRIPTION
**Pull request by fcastilloec**
I have tested this code on both 64 and 32 bits and it's working. I also tested it on different desktop environments and some work better than others, here's a list:

| Desktop | Status |
| --- | --- |
| Unity (Ubuntu <= 17.04) | Works |
| Cinnamon (LInux Mint) | Works |
| Elementary OS | Works |
| Budgie | Can control (No album cover) |
| KDE | Can control (seek is broken [kde bug?]) |
| Gnome Shell | Can Control (but there's no feedback or album cover) |
| Mate | Mate doesn't have mpris support |

I'm not 100% sure why some desktops only have partial support, I think it might be because they're very strict when following the `MPRIS` standard and the nodejs implementation has a big bug [issue #1](https://github.com/emersion/mpris-service/issues/1) which might affect reading metadata to display on their desktops. On the other hand, there's one player using the same `mpris-service` module that doesn't have this problem, and their implementation is almost identical to this.
There's only one bug across all desktops. When changing the volume through `MPRIS` the player doesn't show the change on the screen, even though the volume does change.
We can also implement `loopStatus` and `shuffle` but that will need some work on the player itself, so we 'll need a read/write property for it.

### CODE REVIEW

File: linux/mprisService.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  mprisPlayer.on('seek', (args) => {
+    if (mprisPlayer.playbackStatus == 'Playing' ||
+        mprisPlayer.playbackStatus == 'Paused') {
+      player.webContents.send('win2Player', ['seekTo', args.position/1e6]);
+    }
+  });
+
+  ipcMain.on('win2Player', (e, args) => {
+    if (args[0] == 'playVideo') {
```
Comment (danielravina) [inline linux/mprisService.js]: better use a `switch - case` statement here

**Comment by fcastilloec** [comment]
I fixed the last bug on `mpris`, this is the updated status on the different desktop environments:

| Desktop | Status |
| --- | --- |
| Unity (Ubuntu <= 17.04) | Works |
| Cinnamon (LInux Mint) | Works |
| Elementary OS | Works |
| Gnome Shell | Works |
| Bugie | Can control (No album cover) |
| KDE | Works (seek kde bug) |
| Mate | Mate doesn't have mpris support |

**Comment by fcastilloec** [comment]
FYI, this is the updated status on the different desktop environments:

| Desktop | Status |
| --- | --- |
| Unity (Ubuntu <= 17.04) | Works |
| Cinnamon (LInux Mint) | Works |
| Elementary OS | Works |
| Gnome Shell | Works |
| Budgie | Works |
| KDE | Works |
| Mate | Mate doesn't have mpris support |

File: linux/mprisService.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    if (mprisPlayer.playbackStatus == 'Playing' ||
+        mprisPlayer.playbackStatus == 'Paused') {
+          executeMediaKey(win, 'play-previous')
+    }
+  });
+
+  mprisPlayer.on('volume', (volume) => {
+    if (volume > 1) { volume = 1 }
+    if (volume < 0) { volume = 0 }
+    player.webContents.send('win2Player', ['setVolume', volume*100]);
```
Comment (danielravina) [inline linux/mprisService.js]: There is a problem with this approach. `win2Player` are events that sent from the main window to the youtube player. In this case, It is mpris that is sending the events. The volume bar on the main window will not change. can you confirm?

Comment (fcastilloec) [reply linux/mprisService.js]: Yes, the volume bar doesn't change. I didn't know how to fix that. When sending `seekTo` events in the same way, the main windows does update the current track position, so maybe we could do something similar for the volume.
Another approach would be to execute some javascript directly on the main window that would let us move the volume slider, which will trigger a `setVolume` anyway.
For the first approach, I can't do anything about it. For the second one, just tell me the code to send, and I'll put it there.

Comment (danielravina) [reply linux/mprisService.js]: ok let's merge this and fix it later with a better integration

**Comment by m1guelpf** [comment]
@danielravina Is this ready yet?

**Comment by danielravina** [comment]
it's mergeable unless @fcastilloec wants to add anything else.

**Comment by fcastilloec** [comment]
@danielravina Feel free to merge this PR. I haven't done much work on this lately and the few I've done is not ready to be pushed

**Comment by danielravina** [comment]
Thanks! Sorry for the long delay. awsome contribution!

**Comment by m1guelpf** [comment]
Awesome!

**Pull request merged**